feat: create createBackingVariable method

shivalipatel6 opened the pull request:
Created a protected method that creates the variable for observable_parameter_model to make the class easier to subclass

<!--
  - Thanks for submitting code to Blockly!  Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## The basics

<!-- TODO: Verify the following, checking each box with an 'x' between the brackets: [x] -->

- [X ] I [validated my changes](https://developers.google.com/blockly/guides/contribute/samples#making_and_verifying_a_change)

## The details
### Resolves
<!-- TODO: What Github issue does this resolve? Please include a link. -->
Fixes  https://github.com/google/blockly-samples/issues/1861 


### Proposed Changes
Creates a new method in the ObservableParameterModel class that will be used to create new variables instead of using the constructor. 
<!-- TODO: Describe what this Pull Request does.  Include screenshots if applicable. -->

### Reason for Changes
Makes the class easier to subclass when a developer wants to create a typed variable. 
<!--TODO: Explain why these changes should be made.  Include screenshots if applicable. -->

### Test Coverage
I ran the test class in the block-shareable-procedure files and it ran. 
![createBackingVariableTest](https://github.com/google/blockly-samples/assets/56765670/03f9e9fc-5b29-41a8-8a0b-036a4c111d9d)

<!-- TODO: Please create unit tests, and explain here how they cover
           your changes, or tell us how you tested it manually. If
           your changes include browser-specific behaviour, include
           information about the browser and device that you used for
           testing. -->

### Documentation

<!-- TODO: Does any documentation need to be created or updated because of this PR?
  -        If so please explain.
  -->

### Additional Information

<!-- Anything else we should know? -->

shivalipatel6 committed:
feat!: create createBackingVariable method
Created a protected method that creates the variable for observable_parameter_model to make the class easier to subclass

shivalipatel6 committed:
feat: created a call to createBackingVariable in constructor
This will change my last commit from a breaking change to a non-breaking change.

shivalipatel6 committed:
chore: fix formatting issues
ran npm ci and npm run format to find my formatting errors and fix them

BeksOmega commented on plugins/block-shareable-procedures/src/observable_parameter_model.ts:
```diff
@@ -65,6 +63,21 @@ export class ObservableParameterModel
+   * @returns This parameter model.
+   */
+
+  protected createBackingVariable(name: string, varId?: string): this {
```
Sorry I should have asked for this change in my last review :/

Could you actually change this to return the new variable instead of assigning it? Then you can assign it in the constructor. That way all of the side effects are localized to the constructor, and people overriding this function don't have to remember to do the side effect!

Then this should be good to go =)

shivalipatel6 committed:
fix: change createBackingVariable- return variable
changed the return type of createBackingVariable so that it returns the variable itself.

shivalipatel6 committed:
chore: fix formatting issue
ran npm ci and npm run format to address formatting errors in the pr

BeksOmega commented on plugins/block-shareable-procedures/src/observable_parameter_model.ts:
```diff
@@ -65,6 +63,24 @@ export class ObservableParameterModel
+   * @param varId The optional ID the backing variable should have.
+   * @returns This parameter model.
+   */
+
```
Can you remove this blank line?

BeksOmega commented on plugins/block-shareable-procedures/src/observable_parameter_model.ts:
```diff
@@ -65,6 +63,24 @@ export class ObservableParameterModel
+   *
+   * @param name The string to set set the variable to.
+   * @param varId The optional ID the backing variable should have.
+   * @returns This parameter model.
```
```suggestion
   * @returns The created variable model.
```

BeksOmega reviewed (approved):
Two nits then should be good to go!

shivalipatel6 committed:
chore: fix nits in createBackingVariable issue
